qemu/tests/tpm_check_buffer_size.py

```diff
@@ -0,0 +1,58 @@
+import os
+from shutil import rmtree
+
+from avocado.utils import git
+from avocado.utils import process
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Run Check size of buffer before accessing it in libtpms
+    upstream test tree
+
+    1. Download and install libtpms-devel rpm
+    2. Clone libtpms test tree
+    3. Comple and execute test cases
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    libtpms_rpm = process.getoutput("rpm -q libtpms", shell=True)
+    libtpms_devel_rpm = (libtpms_rpm.split("-", 1)[0] +
+                         "-devel-" + libtpms_rpm.split("-", 1)[1])
+
+    if process.system("rpm -q libtpms-devel", shell=True,
+                      ignore_status=True) != 0:
+        try:
+            process.system(
+                "cd /home/ && brew download-build --rpm %s" %
+                libtpms_devel_rpm, shell=True)
+            libtpms_devel_rpm = libtpms_devel_rpm + ".rpm"
+            process.system("rpm -i /home/%s" % libtpms_devel_rpm, shell=True)
+        except Exception:
+            test.cancel("libtpms-devel package installation failed.")
+
+    repo_url = params["repo_url"]
+    repo_dir = git.get_repo(
+        repo_url, destination_dir=os.path.join(test.tmpdir, "libtpms"))
+
+    try:
+        error_context.base_context(
+            "Build and execute test cases", test.log.info)
+        for test_case in params.get("test_case").split(";"):
+            test_case_o = test_case.split(".")[0]
+            build_execute_cmd = params["build_execute_cmd"] % (
+                test_case, test_case_o, test_case_o)
+            process.system("cd %s && %s" %
+                           (repo_dir, build_execute_cmd), shell=True)
+    finally:
+        rmtree(repo_dir, ignore_errors=True)
+        process.system("rm -f /home/%s" % libtpms_devel_rpm,
```

Comment:
When the system already installed the libtpms-devel pkg, this operation will fail.
```
2023-04-26 21:37:18,238 avocado.utils.process INFO | Running 'rm -f /home/libtpms-devel-0.9.1-3.20211126git1ff6fe1f43.el9_2.aarch64'
2023-04-26 21:37:18,289 avocado.utils.process DEBUG| [stderr] shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
2023-04-26 21:37:18,289 avocado.utils.process INFO | Command 'rm -f /home/libtpms-devel-0.9.1-3.20211126git1ff6fe1f43.el9_2.aarch64' finished with 0 after 0.002187907s
```

And I don't think we need the next line(`rpm -e`), just check if the rpm was installed, if not, then install it.

Reply:
Updated the directory issue.  Thanks.

Reply:
Updated the directory issue.  Thanks.